### ISSUE TITLE
[1.17.1] Player.hasCorrectToolForDrops doesn't check the off-hand

### ISSUE THREAD

**Issue by GizmoTheMoonPig** [MinecraftForge/MinecraftForge#8208]
**Minecraft Version:** 1.17.1

**Forge Version:** 37.0.109

**Steps to Reproduce:**
 1. Create a tool that checks if a block can be harvested (you can use Twilight Forest's Block and Chain as an example)
 2. Use it from your off-hand
 3. Notice how it'll check whatever is in your main hand instead of your off-hand to harvest blocks

**Description of issue:**
Hey there! I'm a Twilight Forest dev and an issue was recently brought to my attention that the block and chain doesn't work as it should in the off-hand. When destroying blocks, we check if the block can actually be harvested with the item. Using Block.canHavestBlock eventually directs you to Player.hasCorrectToolForDrops. This calls for ForgeEventFactory.doPlayerHarvestCheck, and in the success parameter, checks if the selected item is the correct tool. getSelected, unfortunately, only checks the hotbar slots and never the off-hand. This creates a funky issue where it'll ignore the enchantments and harvest level of the block and chain if your main hand is empty, and will inherit the harvest level of the tool in your main hand if you're holding one.

**Comment by diesieben07** [comment]
Can you even mine with your off-hand? When mining a block the main hand is always used...

**Comment by ChampionAsh5357** [comment]
I'd imagine what they are doing dies is creating some sort of item that checks whether you can mine a block via a right-click. So it's more about receiving the correct information on a lookup case compared to actually mining with the off hand.

**Comment by diesieben07** [comment]
Yes, but there is nothing in that method that would allow you to specify which one you want - because it is not necessary in vanilla. Any mod that handles the event will assume it is for the main hand and potentially act incorrectly.

**Comment by GizmoTheMoonPig** [comment]
If there was a way to specify which hand you want to check, that would be nice. Perhaps instead of having this.inventory.getSelected().isCorrectToolForDrops(state) in the doPlayerHarvestCheck boolean, it could be something like this.getItemInHand(hand).isCorrectToolForDrops(state)). While this would solve that issue, I'm not sure how you could easily integrate a hand parameter in without breaking existing code. 
Honestly, when it comes to this I may be the only person who would like a change to this, if you guys dont want to do anything about it I can find another way around it. Just thought I would bring it up for consideration. :)

**Comment by Shadows-of-Fire** [comment]
I don't think harvest checks should be concerned with the off-hand, since breaking is not a supported operation of the off-hand.  If you need to check if the offhand item is able to harvest something during an item use, that would be logic you are responsible for handling as part of the use action.

**Comment by KnightMiner** [comment]
For the record, in the past Tinkers' Construct had offhand harvest. It worked by temporarily putting the tool in the main hand before the block breaking logic, then restoring it to the offhand after (thus automatically having compat with anything assuming harvest comes from main hand). Since the switch happened in the same tick, the client was never updated and the player never noticed.
